rfcs/text/remove-support-for-uncontrolled-inputs.md

```diff
@@ -0,0 +1,233 @@
+- Start Date: 31/03/2022
+
+# Table of contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Basic examples](#basic-examples)
+  - [Component Example - Controlled and Uncontrolled](#component-example---controlled-and-uncontrolled)
+  - [Component Example - Controlled](#component-example---controlled)
+- [Detailed design](#detailed-design)
+  - [Components](#components)
+- [Unit Testing](#unit-testing)  
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Adoption strategy](#adoption-strategy)
+- [How we teach this](#how-we-teach-this)
+- [Documentation](#documentation)
+- [Further Considerations](#further-considerations)
+
+# Summary
+
+The purpose of this document is to outline our intention to remove support for using Carbon input components in an uncontrolled manner.
+
+# Motivation
+
+- To reduce the code complexity in our input components. Currently we have to account for an input being uncontrolled or controlled and therefore have to test these both use cases. We would be able to remove any code relating to invariant messages regarding switching from controlled to uncontrolled or vice versa too.  
+- Uncontrolled behaviour is not the promoted approach to handling inputs in React. React state in their docs that ["In most cases, we recommend using controlled components to implement forms."](https://reactjs.org/docs/uncontrolled-components.html)
+- React form libraries, such as Formik, strongly encourage using controlled components. As can be seen in their docs regarding [using handleChange](https://formik.org/docs/api/formik#handlechange-e-reactchangeeventany--void) and [initialValues](https://formik.org/docs/api/formik#initialvalues-values).
+- To bring the rest of our components inline with the recently refactored Date component which only supports being controlled.
```

Comment:
Should be "in line" rather than "inline". See http://www.kirkmahoney.com/blog/2007/11/in-line-vs-inline/